0033-registry-proxy/README.md

```diff
@@ -0,0 +1,368 @@
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss that.
+-->
+
+Initially, the proxying component will be based on an existing container image using the `socat` binary ([Alpine socat](https://hub.docker.com/r/alpine/socat)). This may evolve to a custom proxy that we create using go. A custom proxy could be useful to run alongside `zarf connect registry` so that tls can be enabled.
+
+`zarf init` will fail if both `--registry-proxy` and `--registry-url` are used. Similarly, init will fail if any of the `--registry-*-file` flags are not empty and `--registry-proxy` is not true.
+
+Running the injector as a daemonset in the airgap requires an image baked into every node, but there is not a distro agnostic way to verify this. The nodeport injector only needs to find one running pod and it can schedule itself on the same node using the same image.
+It is not valid to assume a multi-node cluster has a pod running on each node so the DaemonSet injector cannot use this strategy. However, every node has a field `.status.images`, which Zarf will use to check for the smallest image across every node. Additionally, every node must have access to a pause image, which usually has pause in the name and is extremely small ~300kb. If there's no image shared across nodes Zarf will grab the smallest image with "pause" in the name. If there's no image with "pause" in its name Zarf will fallback to the smallest image on any node.
+
+The long lived injector means that the payload configmaps will no longer be deleted from the cluster during `zarf init --registry-proxy`. This amounts to about an additional 32mb of configmaps stored in the cluster permanently.
+
+In order to allow the Zarf CLI to connect to the internal registry over mTLS Zarf will configure any https requests to the registry to use the CA and client certificates by pulling them from the cluster.
+
+ 
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept proposals with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Zarf testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://docs.zarf.dev/contribute/testing/
+-->
+
+There should be a test that verifies that `zarf init --registry-proxy` works with both NFTables and IPv6. These should both go through the entire e2e suite.
```

Comment:
out-of-scope: would like to see some chaos engineering work its way into these tests - Purposely restart components of the init infrastructure and expect them to recover during package operations.

Reply:
Agreed, or other things like restarting / adding new nodes